x-pack/plugins/ml/public/application/timeseriesexplorer/components/timeseriesexplorer_controls/timeseriesexplorer_controls.tsx

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import React, { useCallback, useState } from 'react';
+import type { EuiContextMenuPanelDescriptor } from '@elastic/eui';
+import {
+  EuiButtonIcon,
+  EuiCheckbox,
+  EuiContextMenu,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiPopover,
+  htmlIdGenerator,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { SaveModalDashboardProps } from '@kbn/presentation-util-plugin/public';
+import {
+  LazySavedObjectSaveModalDashboard,
+  withSuspense,
+} from '@kbn/presentation-util-plugin/public';
+import type { JobId } from '../../../../../common/types/anomaly_detection_jobs/job';
+import { useMlKibana } from '../../../contexts/kibana';
+import { getDefaultSingleMetricViewerPanelTitle } from '../../../../embeddables/single_metric_viewer/get_default_panel_title';
+import type { MlEntity } from '../../../../embeddables';
+import { ANOMALY_SINGLE_METRIC_VIEWER_EMBEDDABLE_TYPE } from '../../../../embeddables/constants';
+import type { SingleMetricViewerEmbeddableState } from '../../../../embeddables/types';
+
+interface Props {
+  selectedDetectorIndex: number;
+  selectedEntities?: MlEntity;
+  selectedJobId: JobId;
+  showAnnotationsCheckbox: boolean;
+  showAnnotations: boolean;
+  showForecastCheckbox: boolean;
+  showForecast: boolean;
+  showModelBoundsCheckbox: boolean;
+  showModelBounds: boolean;
+  toggleShowModelBoundsHandler: () => void;
+  toggleShowAnnotationsHandler: () => void;
+  toggleShowForecastHandler: () => void;
```

Comment:
Shall we follow the naming convention for controlled component? 
```suggestion
  onShowModelBoundsChange: () => void;
  onShowAnnotationsChange: () => void;
  onShowForecastChange: () => void;
```

Reply:
Good call - updated in eacccc621176b8dd78a68c98844e1d5780d4d898